src/com/google/enterprise/gsafeed/Acl.java

```diff
@@ -40,13 +41,45 @@
   @XmlJavaTypeAdapter(NormalizedStringAdapter.class)
   protected String url;
   @XmlAttribute(name = "inheritance-type")
-  @XmlJavaTypeAdapter(CollapsedStringAdapter.class)
-  protected String inheritanceType;
+  protected InheritanceType inheritanceType;
   @XmlAttribute(name = "inherit-from")
   @XmlJavaTypeAdapter(NormalizedStringAdapter.class)
   protected String inheritFrom;
   protected List<Principal> principal;
 
+  /** Inheritance types. */
+  @XmlEnum(String.class)
+  public static enum InheritanceType {
+    @XmlEnumValue("child-overrides")
+    CHILD_OVERRIDES("child-overrides"),
+    @XmlEnumValue("parent-overrides")
+    PARENT_OVERRIDES("parent-overrides"),
+    @XmlEnumValue("and-both-permit")
+    AND_BOTH_PERMIT("and-both-permit"),
+    @XmlEnumValue("leaf-node")
+    LEAF_NODE("leaf-node");
+
+    private String xmlValue;
+
+    private InheritanceType(String xmlValue) {
+      this.xmlValue = xmlValue;
+    }
+
+    @Override
+    public String toString() {
+      return xmlValue;
+    }
+
+    public static InheritanceType fromString(String value) {
+      for (InheritanceType inheritanceType : InheritanceType.values()) {
```

Comment:
In looking at this, we get fromString(null) -> IllegalArgumentException(null). Do we want either fromString(null) -> null (since that's a valid argument to the setters, and it avoids callers who want to use fromString from making the check themselves), or fromString(null) -> NPE?